src/audio/pcm_converter/pcm_converter_generic.c

```diff
@@ -158,12 +158,21 @@ static inline int32_t _pcm_shift(int32_t d, int32_t a)
 
 #define PCM_ABS32(x) (((x) ^ ((x) >> 31)) + (int)((unsigned int)(x) >> 31))
 
-static int32_t _pcm_convert_f_to_i(int32_t src)
+/**
+ * \brief convert float number to fixed point
+ *
+ * \param src integer number to convert
+ * \param pow additional exponent component,
+ *            number of fractional bits in fixed point value.
+ *            Use '0' for normal conversion to integers
+ * \return (float)src * 2**pow
+ */
+static int32_t _pcm_convert_f_to_i(int32_t src, int32_t pow)
 {
 	int32_t ex, mt, dst;
 
 	ex = (src >> 23);
-	ex = (ex & 0xFF) - 127; /* exponential */
+	ex = (ex & 0xFF) + pow - 127; /* exponential */
```

Comment:
This looks like it needs to be squashed into a previous patch.

Reply:
I guess it is easier to follow when conversion float<->fixed is separated with normalization. Moreover there are basically two different operations.